src/main/java/com/amazon/ion/impl/IonReaderContinuableCore.java

```diff
@@ -197,6 +207,19 @@ interface IonReaderContinuableCore extends IonCursor {
     @Deprecated
     int symbolValueId();
 
+    /**
+     * @return true if the value on which the reader is currently positioned is a
+     *  symbol with text available for reading via {@link #getSymbolText()}.
+     */
+    boolean hasSymbolText();
```

Comment:
If `symbolValueId()` returns something less than one, does that always imply `hasSymbolText()` returns `true`? If so, this could be redundant (but that's not necessarily a problem). Either way, it's worth documenting whether there is any relationship between them. (Same goes for `hasFieldText()`.)

Reply:
No; it could have neither a symbol ID nor symbol text, if the reader isn't positioned on a symbol.

Reply:
Added more documentation about the relationships between these methods.

Reply:
> it could have neither a symbol ID nor symbol text, if the reader isn't positioned on a symbol.

Okay, so assuming you are positioned on a symbol value, then a negative SID indicates that there is text. Similarly with field names—if you are in a struct and positioned on a value, then you must be able to get either a non-negative SID or text.